app/experimenter/nimbus-ui/src/hooks/useExperiment.tsx

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useQuery } from "@apollo/client";
+import { GET_EXPERIMENT } from "../gql/experiments";
+import { GetExperiment } from "../gql/types/GetExperiment";
+
+/**
+ * Hook to retrieve all Experiment data to fill in fields on the Edit pages.
```

Comment:
Maybe rephrase this to...

> Hook to retrieve Experiment data by slug.

We'll be using this on the request review page as well as at least the visualization "Results" page.